controlplane/src/core/repositories/SubgraphRepository.ts

```diff
@@ -988,31 +1025,55 @@ export class SubgraphRepository {
       return 0;
     }
 
-    const conditions: SQL<unknown> | undefined = and(
-      // We have this or conditions because we want to fetch the checks based on the new schema or the old schema
-      // as we are not doing a data migration for the checks table
-      or(
-        // This is to fetch the checks based on the new schema
-        inArray(
-          schema.schemaCheckSubgraphs.subgraphId,
-          subgraphs.map(({ id }) => id),
-        ),
-        // This is to fetch the checks based on the old schema
-        inArray(
-          schemaChecks.targetId,
-          subgraphs.map(({ targetId }) => targetId),
-        ),
-      ),
-      gt(schemaChecks.createdAt, new Date(startDate)),
-      lt(schemaChecks.createdAt, new Date(endDate)),
-    );
+    let checks: {
+      id: string;
+    }[] = [];
 
-    // Use a subquery with distinct check IDs to get the count
-    const checks = await this.db
-      .selectDistinct({ id: schemaChecks.id })
-      .from(schemaChecks)
-      .leftJoin(schema.schemaCheckSubgraphs, eq(schema.schemaCheckSubgraphs.schemaCheckId, schemaChecks.id))
-      .where(conditions);
+    // eslint-disable-next-line unicorn/no-negated-condition
+    if (subgraphs.length !== allSubgraphsOfFedGraph.length) {
```

Comment:
This looks complicated for getting the count. Can we simplify that? Even if that means we have to do data migration?

Reply:
Removed the extra query for checks count